src/Compilers/CSharp/Test/Semantic/Semantics/SimpleProgramsTests.cs

```diff
@@ -0,0 +1,1837 @@
+                // (36,38): error CS9000: Cannot use local variable or local function 'Test' declared in a top-level statement in this context.
+                //             System.Console.WriteLine(Test()); // 6
+                Diagnostic(ErrorCode.ERR_SimpleProgramLocalIsReferencedOutsideOfTopLevelStatement, "Test").WithArguments("Test").WithLocation(36, 38),
+                // (37,13): error CS9000: Cannot use local variable or local function 'Test' declared in a top-level statement in this context.
+                //             Test().ToString(); // 7
+                Diagnostic(ErrorCode.ERR_SimpleProgramLocalIsReferencedOutsideOfTopLevelStatement, "Test").WithArguments("Test").WithLocation(37, 13),
+                // (38,13): error CS9000: Cannot use local variable or local function 'Test' declared in a top-level statement in this context.
+                //             Test().EndsWith(null); // 8
+                Diagnostic(ErrorCode.ERR_SimpleProgramLocalIsReferencedOutsideOfTopLevelStatement, "Test").WithArguments("Test").WithLocation(38, 13),
+                // (39,45): error CS9000: Cannot use local variable or local function 'Test' declared in a top-level statement in this context.
+                //             var d = new System.Func<string>(Test); // 9
+                Diagnostic(ErrorCode.ERR_SimpleProgramLocalIsReferencedOutsideOfTopLevelStatement, "Test").WithArguments("Test").WithLocation(39, 45),
+                // (41,24): error CS9000: Cannot use local variable or local function 'Test' declared in a top-level statement in this context.
+                //             _ = nameof(Test); // 10
+                Diagnostic(ErrorCode.ERR_SimpleProgramLocalIsReferencedOutsideOfTopLevelStatement, "Test").WithArguments("Test").WithLocation(41, 24)
+                );
+        }
+
+        [Fact]
+        public void Scope_05()
+        {
+            var text1 = @"
+string Test() => ""1"";
+System.Console.WriteLine(Test());
+";
+            var text2 = @"
+using alias1 = Test;
+
+class Test {}
+
+class Derived : Test
+{
+    void M()
+    {
+        Test x = null;
+        alias1 y = x;
+        System.Console.WriteLine(y);
+        System.Console.WriteLine(Test()); // 1
+        Test().ToString(); // 2
+        Test().EndsWith(null); // 3
+        System.Func<string> d = Test; // 4
+        d();
+        _ = nameof(Test); // 5
+    }
+}
+
+namespace N1
+{
+    using alias2 = Test;
+
+    class Derived : Test
+    {
+        void M()
+        {
+            Test x = null;
+            alias2 y = x;
+            System.Console.WriteLine(y);
+            System.Console.WriteLine(Test()); // 6
+            Test().ToString(); // 7
+            Test().EndsWith(null); // 8
+            var d = new System.Func<string>(Test); // 9
+            d();
+            _ = nameof(Test); // 10
```

Comment:
should there be a `// 10` comment here if there is no diagnostic?

Reply:
> should there be a // 10 comment here if there is no diagnostic?

Same response. Baselines set expectations, not comments in the test snippets.

---
In reply to: [383037440](https://github.com/dotnet/roslyn/pull/41706#discussion_r383037440) [](ancestors = 383037440)